src/hdskins/java/com/voxelmodpack/hdskins/HDSkinManager.java

```diff
@@ -221,78 +213,83 @@ public void skinAvailable() {
             } catch (IOException e) {
                 LogManager.getLogger().trace(e);
             }
-
         }
+
         return textures;
     }
 
     public Map<Type, MinecraftProfileTexture> getProfileData(GameProfile profile) {
         boolean was = !skins.asMap().containsKey(profile);
+
         Map<Type, MinecraftProfileTexture> textures = skins.getUnchecked(profile);
+
         // This is the initial value. Refreshing will load it asynchronously.
         if (was) {
             skins.refresh(profile);
         }
+
         return textures;
     }
 
     public void addSkinServerType(Class<? extends SkinServer> type) {
         Preconditions.checkArgument(!type.isInterface(), "type cannot be an interface");
         Preconditions.checkArgument(!Modifier.isAbstract(type.getModifiers()), "type cannot be abstract");
+
         ServerType st = type.getAnnotation(ServerType.class);
-        if (st == null) {
-            throw new IllegalArgumentException("class is not annotated with @ServerType");
-        }
-        this.skinServerTypes.put(st.value(), type);
+
+        Preconditions.checkArgument(st != null, "class is not annotated with @ServerType");
+
+        skinServerTypes.put(st.value(), type);
     }
 
     public Class<? extends SkinServer> getSkinServerClass(String type) {
-        return this.skinServerTypes.get(type);
+        return skinServerTypes.get(type);
     }
 
     public void addSkinServer(SkinServer skinServer) {
-        this.skinServers.add(skinServer);
+        skinServers.add(skinServer);
     }
 
+    // TODO: Why is this deprecated?
     @Deprecated
     public SkinServer getGatewayServer() {
-        return this.skinServers.get(0);
+        return skinServers.get(0);
     }
 
     public void setEnabled(boolean enabled) {
         this.enabled = enabled;
     }
 
     public static CompletableFuture<PreviewTextureManager> getPreviewTextureManager(GameProfile profile) {
-        return INSTANCE.getGatewayServer().getPreviewTextures(profile).thenApply(PreviewTextureManager::new);
+        return INSTANCE.getGatewayServer().getPreviewTextures(profile).thenApply(mcu -> {
```

Comment:
No. What's hard to understand about `ClassType::new`?

Reply:
This was producing compile errors, so it had to be changed.